### ISSUE TITLE
[JIT] Fix batchnorm / instance norm alias annotations

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#65563]
## 🐛 Bug: mutation to batchnorm weights not reflect in alias analysis

As evidence by the Failing Test [here](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L7947), the alias annotations for batch norm and instance norm are incorrect, because the weight and bias are written to in the operator if `train=True`. We should fix the failing test and improve aliasing handling of the mutation behavior. I would start by updating the [schema](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L749) to reflect the writes. However, we want to preserve the fact that the weights are not written to in inference. I would add special handling of `aten::batchnorm` and `aten::instancenorm` so that we do not register writes either when `train` is False or `is_frozen` is true. Alias Analysis is instantiated with `is_frozen` when we apply optimizations for freezing, and we want to reflect that we'll be in inference mode there even before the `train` parameter is made to be constant. 

You can add special handling [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir/alias_analysis.cpp#L605) and read about the aliasing schema language [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/OVERVIEW.md#handling-mutability).  

You can run the test with `python test/test_ops.py -k test_variant_consistency_jit_nn_functional_batchnorm`.
